src/main/Scheduler.java

```diff
@@ -72,24 +73,28 @@ public void run() {
 					for(DestinationEvent destEvent : destinationEvents) {
 						if(destEvent.getDestination() == arrivalEvent.getCurrentFloor()) {
 							destinationEventFlag = true; 
+							destinationEvents.remove(destEvent);
 							break; 
 						}
 					}
 					
-					//checks if the elevator should keep going. 
-					elevatorKeepsGoing = !destinationEvents.isEmpty();
+					if(!destinationEvents.isEmpty() || floorEventFlag) {
+						//checks if the elevator should keep going. 

```

Comment:
can you quickly fix this too 

elevatorKeepsGoing = (!destinationEvents.isEmpty() || floorEventFlag)

Reply:
acc I can do it when I merge